### DISCUSSION TITLE
Do you need to upload files to the server to use the library?

### DISCUSSION THREAD

**Discussion by jcubic** [naptha/tesseract.js#969] [Q&A]
This looks like a really odd API choice. Is there a way to use File in the browser instead of adding a HTTP server and hosting the image there?

This limitation prevent to use the library without any backend.

**EDIT** just realized that you can create a blob URL, will it work with the library?

**Comment by Balearica** [comment]
I don't know what statement you are referring to.  If you are talking about recognizing [File objects](https://developer.mozilla.org/en-US/docs/Web/API/File) in the browser, this is definitely supported, and multiple of our examples do this (see [here](https://github.com/naptha/tesseract.js/blob/master/examples/browser/basic-efficient.html)).

**Reply by jcubic** [reply]
I was looking at documentation:

https://github.com/naptha/tesseract.js/blob/master/docs/examples.md

It only contains examples with URLs. I would not hide examples like this. You should add a link to more examples from the repo into that markdown file.
